Fails to build with libxkbcommon >= 1.0

**Summary**
Arch AUR fails to build due to `xkb_utf32_to_keysym()` declaration inconsistency

**How to reproduce the issue:**

**Follow Arch AUR install instructions from README.md**
```
sudo pacman -S --needed base-devel git
git clone https://aur.archlinux.org/awesome-git.git
cd awesome-git
makepkg -fsri
```

**Actual result:**

```shell
[ 78%] Building C object CMakeFiles/awesome.dir/luaa.c.o
In file included from /home/joseph/awesome-git/src/build/luaa.c:80:
/home/joseph/awesome-git/src/build/xkb_utf32_to_keysym.c:842:21: error: static declaration of ‘xkb_utf32_to_keysym’ follows non-static declaration
  842 | static xkb_keysym_t xkb_utf32_to_keysym(uint32_t ucs)
      |                     ^~~~~~~~~~~~~~~~~~~
In file included from /home/joseph/awesome-git/src/build/objects/key.h:26,
                 from /home/joseph/awesome-git/src/build/globalconf.h:41,
                 from /home/joseph/awesome-git/src/build/luaa.c:45:
/usr/include/xkbcommon/xkbcommon.h:517:1: note: previous declaration of ‘xkb_utf32_to_keysym’ was here
  517 | xkb_utf32_to_keysym(uint32_t ucs);
      | ^~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/awesome.dir/build.make:186: CMakeFiles/awesome.dir/luaa.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1967: CMakeFiles/awesome.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```

**Expected result:**

(Build completes successfully)

--- a/luaa.c
+++ b/luaa.c
@@ -77,7 +77,7 @@
 
 #include <xcb/xcb_atom.h>
 #include <xcb/xcb_aux.h>
-#include "xkb_utf32_to_keysym.c"
+#include "xkb_utf32_to_keysym_compat.c"
 
 #include <unistd.h> /* for gethostname() */
 
@@ -489,7 +489,7 @@ luaA_get_key_name(lua_State *L)
         keysym = keysyms[0];
     }
     else if ((ucs = one_utf8_to_utf32(input, length)) > 0) //syntax #2
-        keysym = xkb_utf32_to_keysym(ucs);
+        keysym = xkb_utf32_to_keysym_compat(ucs);
     else //syntax #3
         keysym = xkb_keysym_from_name(input, XKB_KEYSYM_NO_FLAGS);
 

--- a/xkb_utf32_to_keysym_compat.c
+++ b/xkb_utf32_to_keysym_compat.c
@@ -839,7 +839,7 @@ static const struct codepair keysymtab[] = {
     { 0x20ac, 0x20ac }, /*                    EuroSign € EURO SIGN */
 };
 
-static xkb_keysym_t xkb_utf32_to_keysym(uint32_t ucs)
+static xkb_keysym_t xkb_utf32_to_keysym_compat(uint32_t ucs)
 {
     /* first check for Latin-1 characters (1:1 mapping) */
     if ((ucs >= 0x0020 && ucs <= 0x007e) ||

